packages/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.h

```diff
@@ -25,6 +25,18 @@ class YogaLayoutableShadowNode : public LayoutableShadowNode {
  public:
   using Shared = std::shared_ptr<const YogaLayoutableShadowNode>;
   using ListOfShared = std::vector<Shared>;
+    
+  /*
+   * Describes a shadow node which is a direct child in the Yoga tree,
+   * not necessarily in the shadow tree. In most cases directChild == yogaChild,
+   * the exception being when directChild has its display set to `contents`.
+   * In that case yogaChild points to a ShadowNode whose yogaNode is a direct
+   * child of this one's.
+   */
+  struct ChildPair {
+    Shared directChild;
+    Shared yogaChild;
+  };
```

Comment:
Could you remind me why we need to store this state, vs deriving it during iteration?